### PR TITLE
Updating kubectl expose to leave service port unnamed

### PR DESCRIPTION
**Pull request by nikhiljindal**
Fixes https://github.com/GoogleCloudPlatform/kubernetes/issues/11377

Introducing a new `service/v2` generator, that leaves the service port unnamed and making that the default generator.
Users can get back the old functionality by passing `--generator=service/v1` to `kubectl expose`

### CODE REVIEW

**Comment by nikhiljindal** [comment]
@bgrant0607

**Comment by bgrant0607** [comment]
LGTM

@jlowdermilk PTAL

File: pkg/kubectl/cmd/expose.go [hunk 1]
```diff
@@ -162,6 +162,10 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		}
 		params["labels"] = kubectl.MakeLabels(labels)
 	}
+	if generatorName == "service/v1" {
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/expose.go]: I feel like this breaks the abstraction of the generator, thus making it less useful. 

As a cleaner approach, can you move the logic inside of [`ServiceGenerator.Generate`](https://github.com/nikhiljindal/kubernetes/blob/kubectlExpose/pkg/kubectl/service.go#L47) to a package private function, then define a `ServiceGeneratorV1` and `ServiceGeneratorV2` whose `Generate` function calls the internal `generate`, setting up extra params as needed? That way clients of the Generator only pass user specified params; any defaults are handled internally according to the Generator version.

Comment (brendandburns) [inline pkg/kubectl/cmd/expose.go]: Yeah, this is exactly why @bgrant0607 made me create the generator stuff in the first place...

Comment (bgrant0607) [inline pkg/kubectl/cmd/expose.go]: That's a good suggestion.

Comment (nikhiljindal) [inline pkg/kubectl/cmd/expose.go]: Thanks for the great suggestion! Updated the code.

**Comment by nikhiljindal** [comment]
Updated the code as per the comment, PTAL

**Comment by jlowdermilk** [comment]
LGTM, thanks

**Comment by nikhiljindal** [comment]
ok-to-merge?

**Comment by bgrant0607** [comment]
cc @ArtfulCoder for merge

**Comment by nikhiljindal** [comment]
Fixed unit tests. All is green now :)

**Comment by bgrant0607** [comment]
Merging

**Comment by bgrant0607** [comment]
Please create a cherrypick for this.

**Comment by nikhiljindal** [comment]
Thanks @bgrant0607 
Sent https://github.com/GoogleCloudPlatform/kubernetes/pull/11433 to request cherrypick.

**Pull request merged**